java-connectors/kafka-connect-gcp-common/src/main/java/io/lenses/streamreactor/connect/gcp/common/auth/GCPServiceBuilderConfigurer.java

```diff
@@ -0,0 +1,93 @@
+import io.lenses.streamreactor.common.config.base.RetryConfig;
+import lombok.experimental.UtilityClass;
+import lombok.val;
+import org.threeten.bp.Duration;
+
+import java.io.IOException;
+import java.util.Optional;
+/**
+ * Utility class for configuring generic GCP service clients using a {@link GCPConnectionConfig}.
+ */
+@UtilityClass
+public class GCPServiceBuilderConfigurer {
+
+  /**
+   * Configures a GCP service client builder with the provided {@link GCPConnectionConfig}.
+   *
+   * @param <X>     Type representing the GCP service interface (e.g., Storage, BigQuery)
+   * @param <Y>     Type representing the service options (e.g., StorageOptions, BigQueryOptions)
+   * @param <B>     Type representing the service options builder (e.g., StorageOptions.Builder, BigQueryOptions.Builder)
+   * @param config  The GCP connection configuration containing settings such as host, project ID, and authentication details.
+   * @param builder The builder instance of the GCP service client options.
+   * @return The configured builder instance with updated settings.
+   * @throws IOException if an error occurs during configuration, such as credential retrieval.
+   */
+  public static <
+          X extends Service<Y>,
+          Y extends ServiceOptions<X, Y>,
+          B extends ServiceOptions.Builder<X, Y, B>
+          >
+  B configure(GCPConnectionConfig config, B builder) throws IOException {
+
+    config.getHost().ifPresent(builder::setHost);
+
+    config.getProjectId().ifPresent(builder :: setProjectId);
+
+    config.getQuotaProjectId().ifPresent(builder :: setQuotaProjectId);
+
+    val authMode = config.getAuthMode();
+
+    builder.setCredentials(authMode.getCredentials());
+
+    builder.setRetrySettings(createRetrySettings(config.getHttpRetryConfig()));
+
+    createTransportOptions(config.getTimeouts()).ifPresent(builder::setTransportOptions);
+
+    return builder;
+  }
+
+  private static Optional<TransportOptions> createTransportOptions(HttpTimeoutConfig timeoutConfig) {
+    if (timeoutConfig.getConnectionTimeout().isPresent() || timeoutConfig.getSocketTimeout().isPresent()) {
+      HttpTransportOptions.Builder httpTransportOptionsBuilder = HttpTransportOptions.newBuilder();
+      timeoutConfig.getSocketTimeout().ifPresent(sock ->
+              httpTransportOptionsBuilder.setReadTimeout(sock.intValue())
+      );
+      timeoutConfig.getConnectionTimeout().ifPresent(conn ->
+              httpTransportOptionsBuilder.setConnectTimeout(conn.intValue())
+      );
+      return Optional.of(httpTransportOptionsBuilder.build());
+    }
+    return Optional.empty();
+  }
+
+  private static RetrySettings createRetrySettings(RetryConfig httpRetryConfig) {
+
+    return RetrySettings.newBuilder()
+            .setInitialRetryDelay(Duration.ofMillis(httpRetryConfig.getErrorRetryInterval()))
+            .setMaxRetryDelay(Duration.ofMillis(httpRetryConfig.getErrorRetryInterval() * 5))
```

Comment:
This is the existing behaviour (in the scala code) so I'm going to leave it as it is, the point of this PR is to translate the existing functionality into java.

Ideally, you're right, but at the same time we can't expose every single property of every connector, as configuration becomes overcomplex and also bear in mind the connectors are not equivalent (RetryConfig is a leaky abstraction - if we add a `maxRetryInterval` to `RetryConfig` then this will not be relevant everywhere that `RetryConfig` is used).  i hope you don't disagree that the multiplier is an acceptable default for now.